src/DependencyInjection/Providers/AmazonProviderConfigurator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * OAuth2 Client Bundle
+ * Copyright (c) KnpUniversity <http://knpuniversity.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace KnpU\OAuth2ClientBundle\DependencyInjection\Providers;
+
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+
+class AmazonProviderConfigurator implements ProviderConfiguratorInterface
+{
+    public function buildConfiguration(NodeBuilder $node)
+    {
+        // no custom options
+    }
+
+    public function getProviderClass(array $config)
+    {
+        return 'Luchianenco\OAuth2\Client\Provider\Amazon';
```

Comment:
I'm just wondering why not using the syntax `Amazon::class`?

Reply:
well, I tried to keep the style the repository had before ;)
of course it can be a next PR ;) to change in all Configurators

Reply:
Oh, yeah, maybe @weaverryan can tell us the true reason behind this :)

Reply:
I think it's just legacy from the past, but since we require PHP `>=5.5.9` in `composer.json` now changing to `::class` seems reasonable to me, but yes, in another PR and for all occurrences 👍

Reply:
Uh oh, probably I was wrong! The main problem is that we do not have strict dependencies on those classes, i.e. user installs them if needed. So, it'll throw an error if the class does not found. That's why here we use strings which postpone this check until actual call. Sorry for misleading you in my previous message 😇

Reply:
Ahh, that makes sense, thanks for the clarification